covid_workplace_model.py

```diff
@@ -0,0 +1,461 @@
+            probability_contacts_sustain : probability that contacts on consecutive days remain
+                                            the same as the previous day. 1 implies that
+                                            all contacts stay the same on consecutive days,
+                                            while 0 implies that previous day's contacts have
+                                            no consequence on next day's contacts.
+            sar_average : average secondary attack rate in range [0, 1]
+            case_rate : daily new cases per population, common ranges are 0.00001 to 0.001
+            num_days : the number of days over which to estimate incidence in the workplace
+            num_sims : number of simulations of stochastic model
+            output_file : filename where csv output will be saved, if provided
+    -----Returns-----
+            None
+    """
+    if output_file is not None:
+        cumulative_incidence = calculate_nday_workplace_incidence(
+            employees_at_work=employees_at_work,
+            case_rate=case_rate,
+            daily_contact_size=daily_contact_size,
+            probability_contacts_sustain=probability_contacts_sustain,
+            sar_average=sar_average,
+            num_days=num_days,
+            num_sims=num_sims,
+        )
+        logging.info(
+            f"Calculated cumulative incidence for employees:{employees_at_work}, "
+            f"case_rate:{case_rate}, contacts:{daily_contact_size}, sar:{sar_average}"
+        )
+        csv_header = [f"day_{x+1}" for x in range(num_days)]
+        csv_header = ",".join(csv_header)
+        logging.info(f"Writing cumulative incidence to file:{output_file}")
+        np.savetxt(
+            output_file,
+            cumulative_incidence,
+            fmt="%d",
+            delimiter=",",
+            header=csv_header,
+            comments="",
+        )
+
+
+def main(args=None):
+    if args is not None:
+        tocsv_nday_workplace_incidence(
+            employees_at_work=args.employees,
+            case_rate=args.caserate,
+            daily_contact_size=args.contacts,
+            probability_contacts_sustain=args.prob_sustain,
+            sar_average=args.sar,
+            num_days=args.days,
+            num_sims=args.sims,
+            output_file=args.outfile,
+        )
+
+
+if __name__ == "__main__":
+    # execute only if run as a script
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Calculate cumulative incidence for COVID-19 in workplaces"
+    )
+    parser.add_argument(
+        "-e",
+        "--employees",
+        type=int,
+        metavar="INTEGER>0",
+        required=True,
+        help="number of employees in the workplace",
+    )
+    parser.add_argument(
+        "-c",
+        "--caserate",
+        type=float,
+        metavar="[0., 1.]",
+        required=True,
+        help="daily new cases per population",
```

Comment:
suggestion on changing this to per 100k population, right?

Reply:
No, because units are per population. So if cases are 1000 per 100k population, value here is 0.01.